rca/projects/models.py

```diff
@@ -265,4 +273,164 @@ def get_context(self, request, *args, **kwargs):
+                    "link": slugify(i.title),
+                    "filter_type": "school_or_centre",
+                }
+            )
+        filters.append(school_or_centre)
+
+        return filters
+
+    def _format_results(self, projects):
+        """ Prepares the queryset into a digestable list for the template """
+        projects_formatted = []
+        for page in projects:
+            year = None
+            if page.start_date:
+                year = page.start_date.strftime("%Y")
+                if page.end_date and page.end_date.strftime("%Y") != year:
+                    end_year = page.end_date.strftime("%Y")
+                    year = year + " - " + end_year
+
+            projects_formatted.append(
+                {
+                    "title": page.title,
+                    "image": page.hero_image,
+                    "link": page.url,
+                    "school": page.get_related_school(),
+                    "year": year,
+                }
+            )
+        return projects_formatted
+
+    def get_extra_query_params(self, request):
+        # Formats the request parameters in a list
+        extra_query_params = []
+        for i in request.GET.getlist("type"):
+            extra_query_params.append(urlencode({"type": i}))
+        for i in request.GET.getlist("subject"):
+            extra_query_params.append(urlencode({"subject": i}))
+        for i in request.GET.getlist("school_or_centre"):
+            extra_query_params.append(urlencode({"school_or_centre": i}))
+        return extra_query_params
+
+    def get_results(self, request, context):
+        projects = (
+            ProjectPage.objects.live()
+            .public()
+            .descendant_of(self, inclusive=True)
+            .select_related("hero_image")
+        )
+
+        # Request filters
+        research_types = request.GET.getlist("type")
+        subjects = request.GET.getlist("subject")
+        school_or_centre = request.GET.getlist("school_or_centre")
+
+        if research_types:
+            projects = projects.filter(
+                research_types__research_type_id__in=research_types
+            )
+
+        if subjects:
+            projects = projects.filter(subjects__subject_id__in=subjects)
+
+        if school_or_centre:
+            projects = projects.filter(
+                models.Q(related_school_pages__page_id__in=school_or_centre)
+                | models.Q(related_research_pages__page_id__in=school_or_centre)
+            )
+        return self._format_results(projects.distinct())
+
+    def get_context(self, request, *args, **kwargs):
+        page = request.GET.get("page", 1)
+        context = super().get_context(request, *args, **kwargs)
+        context["filters"] = self.get_filters()
+        context["featured_project"] = self.featured_project
+
+        project_results = self.get_results(request, context)
+        context["results_count"] = len(project_results)
```

Comment:
The paginator objects provides a `.count` attribute, could that be used instead?
https://docs.djangoproject.com/en/3.0/topics/pagination/#example